astroid/__pkginfo__.py

```diff
@@ -15,7 +21,26 @@
 
 extras_require = {}
 install_requires = ['lazy_object_proxy', 'six', 'wrapt']
-extras_require[':python_version<"3.4"'] = ['enum34', 'singledispatch']
+
+
+def has_environment_marker_range_operators_support():
+    """Code extracted from 'pytest/setup.py'
+    https://github.com/pytest-dev/pytest/blob/7538680c/setup.py#L31
+
+    The first known release to support environment marker with range operators
+    it is 17.1, see: https://setuptools.readthedocs.io/en/latest/history.html#id113
+    """
+    try:
+        return parse_version(setuptools_version) >= parse_version('17.1')
+    except Exception:
```

Comment:
I am pretty sure then this can be simplified. You can try to investigate what errors parse_version is supposed to raise and catch only those, for instance.